tests/integration-tests/tests/sqlite_authority_tests.rs

```diff
@@ -954,9 +954,13 @@ async fn test_zone_signing() {
         assert!(
             inner_results
                 .iter()
-                .filter(|r| r.record_type() == RecordType::RRSIG && r.name() == record.name())
-                .filter_map(Record::data)
-                .filter_map(RRSIG::try_borrow)
+                .filter_map(|r| {
```

Comment:
ok, @djc, fully combined... I guess I don't see this as "cleaner" but it does keep the logic contained. I'm ok with this construct, if you like it, we can adopt it as a standard across the code base.